scripts/maint/rename_c_identifier.py

```diff
@@ -0,0 +1,214 @@
+        qid2 = shlex.quote(id2)
+        script.append("        {} {}".format(qid1,qid2))
+    script = " \\\n".join(script)
+
+    if len(pairs) == 1:
+        line1 = "Rename {} to {}".format(pairs[0])
+    else:
+        line1 = "Replace several C identifiers."
+
+    msg = """\
+{}
+
+This is an automated commit, generated by this command:
+
+{}
+""".format(line1, script)
+
+    if no_verify:
+        msg += """
+It was generated with --no-verify, so it probably breaks some commit hooks.
+The commiter should be sure to fix them up in a subsequent commit.
+"""
+
+    return msg
+
+def commit(pairs, no_verify=False):
+    """Try to commit the current git state, generating the commit message as
+       appropriate.  If `no_verify` is True, pass the --no-verify argument to
+       git commit.
+    """
+    args = []
+    if no_verify:
+        args.append("--no-verify")
+
+    fname = None
+    try:
+        with tempfile.NamedTemporaryFile(mode="w",delete=False) as f:
+            fname = f.name
+            f.write(make_commit_msg(pairs, no_verify))
+        s = subprocess.run(["git", "commit", "-a", "-F", fname, "--edit"]+args)
+        if s.returncode != 0 and not no_verify:
+            print('"git commit" failed. Maybe retry with --no-verify?',
+                  file=sys.stderr)
+            revert_changes()
+            return False
+    finally:
+        os.unlink(fname)
+
+    return True
+
+def any_uncommitted_changes():
+    """Return True if git says there are any uncommitted changes in the current
+       working tree; false otherwise.
+    """
+    s = subprocess.run(["git", "diff-index", "--quiet", "HEAD"])
+    return s.returncode != 0
+
+def revert_changes():
+    """Tell git to revert all the changes in the current working tree.
+    """
+    print('Reverting changes.',file=sys.stderr)
+    subprocess.run(["git", "checkout", "--quiet", TOPDIR])
+
+def main(argv):
+    import argparse
+    parser = argparse.ArgumentParser(description="Rename a C identifier.")
+    parser.add_argument("--no-verify", action='store_true')
+    parser.add_argument("--commit", action='store_true')
+    parser.add_argument("from_id", type=str,  help="Original identifier")
+    parser.add_argument("to_id", type=str, help="New identifier")
+    parser.add_argument("more", type=str, nargs=argparse.REMAINDER,
+                        help="Additional identifier pairs")
+
+    args = parser.parse_args(argv[1:])
+
+    if len(args.more) % 2 != 0:
+        print("I require an even number of identifiers.",file=sys.stderr)
+        return 1
+
+    if any_uncommitted_changes():
+        print("Uncommitted changes found. Not running.", file=sys.stderr)
+        return 1
+
+    pairs = []
+    print("renaming {} to {}".format(args.from_id, args.to_id))
+    pairs.append((args.from_id, args.to_id))
+    for idx in range(0,len(args.more),2):
+        id1 = args.more[idx]
+        id2 = args.more[idx+1]
+        print("renaming {} to {}".format(id1, id2))
```

Comment:
Not sys.stderr like other prints

Reply:
c69f17ae28fdc626e3019468a62c9dce4915b680